src/api/index.ts

```diff
@@ -0,0 +1,157 @@
+import Database, { Database as DatabaseInstance } from 'better-sqlite3'
+import { FastifyInstance, FastifyPluginAsync, FastifyRequest } from 'fastify'
+import fp from 'fastify-plugin'
+import path from 'path'
+import Piscina from 'piscina'
+import { MessagePort } from 'worker_threads'
+
+import { convertActiveToError as convertActiveImportsToErrorImports } from '../lib/imports'
+import { migrate } from '../lib/migrations'
+import { UpstreamRequestsManager } from '../lib/upstream_requests_manager'
+import createImportsApi, { ImportsApi } from './imports'
+import createSpritesApi, { SpritesApi } from './sprites'
+import createStylesApi, { StylesApi } from './styles'
+import createTilesApi, { TilesApi } from './tiles'
+import createTilesetsApi, { TilesetsApi } from './tilesets'
+
+export interface MapServerOptions {
+  dbPath: string
+}
+
+export interface Context {
+  activeImports: Map<string, MessagePort>
+  db: DatabaseInstance
+  piscina: Piscina
+  upstreamRequestsManager: UpstreamRequestsManager
+}
+
+// Any resource returned by the API will always have an `id` property
+export interface IdResource {
+  id: string
+}
+
+export interface Api
+  extends ImportsApi,
+    SpritesApi,
+    StylesApi,
+    TilesApi,
+    TilesetsApi {}
+
+function createApi(context: Context): Api {
+  const tilesetsApi = createTilesetsApi({ context })
+
+  const stylesApi = createStylesApi({
+    api: {
+      createTileset: tilesetsApi.createTileset,
+    },
+    context,
+  })
+
+  const importsApi = createImportsApi({
+    api: {
+      createTileset: tilesetsApi.createTileset,
+      createStyleForTileset: stylesApi.createStyleForTileset,
```

Comment:
It's a bit clunky, but I think this is the best solution for inter-dependencies between parts of the API. Hopefully we don't need anything circular! I like that with this approach the dependencies are explicit.